File: frontend/erp-portfolio-web/src/types/vendedor.ts
/**
 * =====================================================================
 * Arquivo....: vendedor.ts
 * Versão.....: 1.0.0
 * Data.......: 23/09/2026
 * Descrição..: Tipos do cadastro de vendedores, espelhando os DTOs da API
 *              (VendedorRespostaDto, VendedorCriacaoDto, VendedorAtualizacaoDto,
 *              VendedorFiltroDto).
 * ---------------------------------------------------------------------
 * Histórico de alterações:
 *   1.0.0 - 23/09/2026 - Criação do arquivo.
 * =====================================================================
 */

export interface Vendedor {
  id: number
  nome: string
  /** 11 dígitos, sem máscara. */
  cpf: string
  email: string | null
  telefone: string | null
  /** % de comissão padrão (0 a 100), copiada para o pedido ao confirmar. */
  percentualComissao: number
  ativo: boolean
  /** Data/hora ISO 8601 em UTC. */
  dataCadastro: string
}

export interface VendedorAtualizacao {
  nome: string
  cpf: string
  email: string | null
  telefone: string | null
  percentualComissao: number
  ativo: boolean
}

export interface VendedorFiltro {
  nome?: string
  /** true = só ativos, false = só inativos, ausente = todos. */
  ativo?: boolean
  pagina: number
  tamanhoPagina: number
}

File: backend/ErpPortfolio.Api/Models/Vendedor.cs
// =====================================================================================
// Arquivo....: Vendedor.cs
// Versão.....: 1.0.0
// Data.......: 23/09/2026
// Descrição..: Entidade de domínio que representa um vendedor (pessoa física), com a %
//              de comissão padrão. A % é copiada para o pedido ao confirmar (SPEC.md, PV3).
// -------------------------------------------------------------------------------------
// Banco......: PostgreSQL - erp_portfolio_db
// Tabelas....: public.vendedores
// Fontes.....: Mapeada em ErpPortfolioDbContext.Vendedores (EF Core / Npgsql).
//              Colunas: id, nome, cpf, email, telefone, percentual_comissao, ativo,
//              data_cadastro (o mapeamento de nomes fica no DbContext).
// -------------------------------------------------------------------------------------
// Histórico de alterações:
//   1.0.0 - 23/09/2026 - Criação do arquivo.
// =====================================================================================

namespace ErpPortfolio.Api.Models;

public class Vendedor
{
    public int Id { get; set; }

    public string Nome { get; set; } = string.Empty;

    /// <summary>CPF com 11 dígitos, armazenado sem máscara.</summary>
    public string Cpf { get; set; } = string.Empty;

    public string? Email { get; set; }

    public string? Telefone { get; set; }

    /// <summary>% de comissão padrão (0 a 100), copiada para o pedido ao confirmar.</summary>
    public decimal PercentualComissao { get; set; }

    public bool Ativo { get; set; } = true;

    /// <summary>Data/hora de cadastro em UTC.</summary>
    public DateTime DataCadastro { get; set; }
}
